astropy/coordinates/angles.py

```diff
@@ -644,12 +650,20 @@ def _wrap_internal(self):
         # Convert the wrap angle and 360 degrees to the native unit of
         # this Angle, then do all the math on raw Numpy arrays rather
         # than Quantity objects for speed.
+        # TODO: This could probably be sped up as well.  u.degree.to(self.unit)
```

Comment:
FWIW the if statement below accounts for about 3.5% of the runtime in the original test case.  So, not enormous.  I was just trying to shave off what I can.  What do you think?  Worth leaving in, or better to just wait and make Unit faster for this case?